### PR TITLE
Native param types

### PR DESCRIPTION
**Pull request by kamil-tekiela**
3/3. Adding native param types was the most challenging of native types. I caught and fixed a lot of issues but I am sure I missed many more. 

Prerequisities:
- #18244
- #18218

### CODE REVIEW

**Review by williamdes — APPROVED** [review]
🥇 🥇 🥇

**Comment by kamil-tekiela** [comment]
The transformation tests are failing on ARM but I don't know why.

**Comment by MauricioFauth** [comment]
> The transformation tests are failing on ARM but I don't know why. 

Looks like a 32 bit issue.

**Comment by kamil-tekiela** [comment]
> > The transformation tests are failing on ARM but I don't know why.
> 
> Looks like a 32 bit issue.

Is it because of my PR? How can I fix this?

File: libraries/classes/Plugins/Transformations/Abs/LongToIPv4TransformationsPlugin.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public static function getInfo(): string
      * @param array              $options transformation options
      * @param FieldMetadata|null $meta    meta information
      */
-    public function applyTransformation($buffer, array $options = [], FieldMetadata|null $meta = null): string
+    public function applyTransformation(string $buffer, array $options = [], FieldMetadata|null $meta = null): string
```
Comment (MauricioFauth) [inline libraries/classes/Plugins/Transformations/Abs/LongToIPv4TransformationsPlugin.php:40]: Not sure, but I think is this line that is causing the test to fail.

Comment (MauricioFauth) [reply libraries/classes/Plugins/Transformations/Abs/LongToIPv4TransformationsPlugin.php:40]: Looks like ReflectionMethod::invokeArgs does not raise a type error. 

https://3v4l.org/MBVNG

Comment (kamil-tekiela) [reply libraries/classes/Plugins/Transformations/Abs/LongToIPv4TransformationsPlugin.php:40]: I fixed it but it still fails. I don't think that was it. I think I will take this change out of this PR.

**Review by MauricioFauth — APPROVED** [review]

**Comment by MauricioFauth** [comment]
:100:

**Pull request merged**